Framework/Algorithms/src/PDFFourierTransform.cpp

```diff
@@ -220,50 +306,24 @@ void PDFFourierTransform::exec() {
   }
 
   // determine Q-range
-  double qmin = getProperty("Qmin");
-  double qmax = getProperty("Qmax");
-  if (isEmpty(qmin))
-    qmin = inputQ.front();
-  if (isEmpty(qmax))
-    qmax = inputQ.back();
-
-  // check Q-range and print information
-  if (qmin < inputQ.front()) {
-    g_log.information()
-        << "Specified Qmin < range of data. Adjusting to data range.\n";
-    qmin = inputQ.front();
-  }
-  if (qmax > inputQ.back()) {
-    g_log.information()
-        << "Specified Qmax > range of data. Adjusting to data range.\n";
-  }
-  g_log.debug() << "User specified Qmin = " << qmin
-                << "Angstroms^-1 and Qmax = " << qmax << "Angstroms^-1\n";
-
-  // get pointers for the data range
-  size_t qmin_index;
-  size_t qmax_index;
+  size_t qmin_index = determineQminIndex(inputQ, inputFOfQ);
+  size_t qmax_index = determineQmaxIndex(inputQ, inputFOfQ);
   { // keep variable scope small
-    auto qmin_ptr = std::upper_bound(inputQ.begin(), inputQ.end(), qmin);
-    qmin_index = std::distance(inputQ.begin(), qmin_ptr);
-    if (qmin_index == 0)
-      qmin_index += 1; // so there doesn't have to be a check below
-    auto qmax_ptr = std::lower_bound(inputQ.begin(), inputQ.end(), qmax);
-    qmax_index = std::distance(inputQ.begin(), qmax_ptr);
+    size_t qmi_out = qmax_index;
+    if (qmi_out == inputQ.size())
+      qmi_out--; // prevent unit test problem under windows (and probably other
+    // hardly identified problem)
```

Comment:
Is this mess still necessary? Your new `determineQMaxIndex` seems to return only valid `qmax_index` values?